pkg/qrcodegenerator/qrcode/enddevices/enddevices.go

```diff
@@ -0,0 +1,123 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package enddevices
+
+import (
+	"context"
+	"sync"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/qrcodegenerator/qrcode"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+const (
+	serialNumberAttribute = "serial-number"
+	vendorIDAttribute     = "vendor-id"
+	profileIDAttribute    = "profile-id"
```

Comment:
I think this need to become `EndDevice` and/or `EndDeviceVersionIdentifiers` fields